StringPrototypeExtension.js

```diff
@@ -0,0 +1,185 @@
+String.prototype.hasVowels = function() {
+  var vowelRegex = /[aeiou]/gi;
+  var hasVowel = vowelRegex.test(this);
+  return hasVowel;
+};
+
+String.prototype.isQuestion = function() {
+  var isQuestionRegex = /\w+[?]$/;
+  return isQuestionRegex.test(this);
+};
+
+String.prototype.numberWords = function() {
+  var numberWordRegex = [
+    "zero",
+    "one",
+    "two",
+    "three",
+    "four",
+    "five",
+    "six",
+    "seven",
+    "eight",
+    "nine"
+  ];
+  var digitsArray = this.match(/\d/g);
+  var numberToWord = "";
+  digitsArray.forEach(digit => {
+    numberToWord += numberWordRegex[digit].concat(" ");
+  });
+  return numberToWord;
+};
+
+String.prototype.wordsToArray = function() {
+  var wordRegex = /\b[0-9a-zA-Z_\-?]+\b/g;
+  var wordArray = this.match(wordRegex);
+
+  return wordArray;
+};
+
+String.prototype.wordCount = function() {
+  return this.wordsToArray().length;
+};
+
+String.prototype.isDigit = function() {
+  var digitRegex = /\d/g;
+
+  if (!digitRegex.test(this)) {
```

Comment:
Refactoring this if statement to one line of code would look more elegant. Please remove the braces, put the return statement on the same line as the if statement and add a comment above that line to explain what it is doing.